src/Component/Field.js

```diff
@@ -0,0 +1,275 @@
+import React,{useState} from 'react'
+
+function Field({fieldData}) {
+
+    const [text,setText] = useState('')
+    const [fieldValue,setFieldValue] = useState([]);
+    const [finalFormData, setFinalFormData] = useState([])
+
+    let allFields = [];
+    let element = ''
+
+    function handlerButtonTypeButton(){
+      console.log("Button Type Button Clicked !")
+    }
+
+    function handleButtonTypesubmit(){
+         console.log("Button Type Submit  clicked !")
+         setFinalFormData(fieldValue.map(field => field));
+    } 
+   
+    function handleButtonTypeReset() {
+        setText('')
+        setFinalFormData([]);
+    }
+
+
+   function fieldClickhandler(e) {
+        switch(e.target.type){
+          case 'button': 
+             handlerButtonTypeButton();
+             break;
+           case 'submit': 
+             handleButtonTypesubmit();
+             break;
+           case "reset":
+             handleButtonTypeReset();
+             break;
+        }
+   }
+
+    function handleOnchange(e,field) {
+        setText(e.target.value)
+    } 
+
+    function handleBulr(e,field){ 
+      console.log(e.target)
+      //apply validation here 
+        if(e.target.value==="") return
+        if(fieldValue.length===0) return setFieldValue([...fieldValue,{name:field.attributes.name,value:e.target.value}])
```

Comment:
If fieldValue length is 0 then no need to have `[...fieldValue,{name:field.attributes.name,value:e.target.value}]`
You can have instead `[{name:field.attributes.name,value:e.target.value}]`

Reply:
Yes